src/RpcClient.ts

```diff
@@ -630,32 +636,44 @@ export class RpcClient {
    * @param {TransactionInstruction[]} instructions - The transaction instructions
    * @param {Signer[]} signers - The transaction's signers. The first signer should be the fee payer
    * @param {AddressLookupTableAccount[]} lookupTables - The lookup tables to be included in a versioned transaction. Defaults to `[]`
-   * @param {SendOptions} sendOptions - Options for sending the transaction. Defaults to `{ skipPreflight: false }`
+   * @param {SendOptions} sendOptions - Options for sending the transaction
    * @returns {Promise<TransactionSignature>} - The transaction signature
   */
   async sendSmartTransaction(
     instructions: TransactionInstruction[],
     signers: Signer[],
     lookupTables: AddressLookupTableAccount[] = [],
-    sendOptions: SendOptions = { skipPreflight: false },
+    sendOptions?: SendOptions,
   ): Promise<TransactionSignature> {
     try {
       // Create a smart transaction
-      const transaction = await this.createSmartTransaction(instructions, signers, lookupTables, sendOptions);
+      const { txBuff, blockhash, minContextSlot } = await this.createSmartTransaction(instructions, signers, lookupTables);
+
+      const recdSendOptions: SendOptions = {
+        maxRetries: 0,
+        preflightCommitment: "confirmed",
+        skipPreflight: true,
+        minContextSlot: minContextSlot,
+      };
+      const commitment = sendOptions?.preflightCommitment || recdSendOptions.preflightCommitment;
   
       // Timeout of 60s. The transaction will be routed through our staked connections and should be confirmed by then
       const timeout = 60000;
       const startTime = Date.now();
-      let txtSig;
   
       while (Date.now() - startTime < timeout) {
         try {
-          txtSig = await this.connection.sendRawTransaction(transaction.serialize(), {
-            skipPreflight: sendOptions.skipPreflight,
+          // signature does not change when it resends the same one
+          const signature = await this.connection.sendRawTransaction(txBuff, {
+            ...recdSendOptions,
             ...sendOptions,
           });
   
-          return await this.pollTransactionConfirmation(txtSig);
+          const abortSignal = AbortSignal.timeout(15000);
+          await this.connection.confirmTransaction({ abortSignal, signature, ...blockhash }, commitment);
```

Comment:
@0xIchigo Do you see any issue getting use of default `confirmTransaction` method instead of recreating `pollTransactionConfirmation`?

Reply:
Have you tested sending transactions with this change? The goal of `pollTransactionConfirmation` is to check whether the transaction has been confirmed over the last x seconds (right now set to 15) and then retry the transaction if it isn't confirmed. My concern with using `confirmTransaction` is the underlying code already includes timeout and polling logic that we can't really abstract out and change in the future

Reply:
AbortSignal allows you to inject custom timeout logic for the confirm transaction. I was able to resend raw transaction after 15 seconds with this codebase.

Reply:
So, just to confirm, the transaction is polled continuously for 15 seconds, and gets retried if it fails up until that 60s timeout?

Reply:
Yes, it works like you described.

Reply:
Sounds good! I have a few other PRs that I've just opened that need to be merged first, so we may need to refactor this one a bit. Maybe we could still keep `pollTransactionConfirmation` but make it more modular so developers can use it at their own discretion, and then implement `confirmTransaction` for the smart transactions?